### PR TITLE
Support custom migration tables

### LINKED ISSUE #3213: Make phinx support modules
Allow modules to use the phinx system 

Example as @Aberdeener mentioned
```php
class Module {
	public static function migrationDirs() {
	  $dirs = [];
	  $modules = Module::getModules();
	  foreach ($module as $module) {
	    if (dir_exists($migration_path = '/path/to/$module/migrations') {
	      $dirs[] = $migration_path;
	    }
	  }
	
	  return $dirs;
	}
}

// then in phinx.php
'paths' => [
  'migrations' => [__DIR__, ...Module::migrationDirs()],
],
```

### PR DESCRIPTION
**Pull request by samerton**
Closes #3213

### CODE REVIEW

File: core/classes/Database/PhinxAdapter.php [hunk 1]
```diff
@@ -73,4 +121,52 @@ public static function migrate(): string {
...
+     *
+     * @param string $module Module name
+     * @param string $migrationDir Migration directory to use
+     * @param int $since Version of earliest migration to rollback, default 0 for all
+     *
+     * @return string Output of the migration command from Phinx as if it was executed in the console.
+     *
+     * @throws Exception If unable to rollback
+     */
+    public static function rollback(
```
Comment (tadhgboyle) [inline core/classes/Database/PhinxAdapter.php:136]: Good addition - I wonder where/how we could implement this

**Review by tadhgboyle — APPROVED** [review]
Looks good! We should remember to update the docs to show a quick guide on how module devs can use this

File: core/classes/Database/PhinxAdapter.php (hunk 1 above)
Comment (samerton) [reply core/classes/Database/PhinxAdapter.php:136]: The use case I had in mind was for `onUninstall` for modules - we can remove the database tables when uninstalling a module

**Comment by partydragen** [comment]
HYPE!

**Pull request merged**